### PR TITLE
Refactor internals

### PR DESCRIPTION
**Pull request by seizans**
I refactored internal codes.
(e.g. to use the standard URI module and URI struct)

If it is good, I hope you will merge it.

### CODE REVIEW

File: lib/scrivener/headers.ex [hunk 1]
```diff
@@ -17,71 +17,56 @@ defmodule Scrivener.Headers do
...
-  defp page_link(conn, %{page_number: page}, "next"), do: page_link(conn, page + 1, "next")
-
-  defp page_link(_conn, %{page_number: 1}, "prev"), do: ""
-  defp page_link(conn, %{page_number: page}, "prev"), do: page_link(conn, page - 1, "prev")
-
-  defp page_link(conn, page, rel) when is_integer(page) do
-    conn
-    |> paged_url(page)
-    |> page_link(rel)
+  defp maybe_add_prev(links, uri, page_number, total_pages) do
```
Comment (doomspork) [inline lib/scrivener/headers.ex]: Let's drop the `if/else` and use guards for both `maybe_add_prev/4` and `maybe_add_next/4`

File: lib/scrivener/headers.ex [hunk 2]
```diff
@@ -17,71 +17,56 @@ defmodule Scrivener.Headers do
...
+    |> maybe_add_next(uri, page.page_number, page.total_pages)
+    |> Enum.join(", ")
   end
 
-  defp pages(conn, page) do
-    @rels
-    |> Enum.map(&(page_link(conn, page, &1)))
-    |> Enum.reject(&(&1 == ""))
+  defp link_str(%{query: req_query} = uri, page_number, rel) do
+    query = req_query
```
Comment (doomspork) [inline lib/scrivener/headers.ex]: Let's move these pipes to a new line:

```elixir
query = 
  req_query
  |> URI.decode_query()
  |> Map.put("page", page_number)
  |> URI.encode_query()
```

**Comment by doomspork** [comment]
Thanks @seizans!  I've left some feedback for you to consider

File: lib/scrivener/headers.ex (hunk 2 above)
Comment (seizans) [inline lib/scrivener/headers.ex]: Should I update pipes only here?
I think there are more relevant parts and it's better to unify styles.

```
  Use `paginate/2` to set the pagination headers:
      def index(conn, params) do
        page = MyApp.Person
               |> where([p], p.age > 30)
               |> order_by([p], desc: p.age)
               |> preload(:friends)
               |> MyApp.Repo.paginate(params)
```

```
    uri = %URI{scheme: Atom.to_string(conn.scheme),
               host: conn.host,
               port: conn.port,
               path: conn.request_path,
               query: conn.query_string}
```

```
    uri_str = %URI{uri | query: query}
              |> URI.to_string()
```

Comment (doomspork) [inline lib/scrivener/headers.ex]: Feel free to change any of the pipes within your change but I'd prefer to not go adding these changes to files unrelated to these changes, we'll get them when there is a need to update those files.

**Comment by seizans** [comment]
@doomspork 
Thank you for your kind feed back.
I fixed them.
Can you check it please?

**Comment by doomspork** [comment]
Thank you @seizans 👍

**Pull request merged**